packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2442,6 +2444,31 @@ export class MergeTree {
         }
     }
 
+    public removeLocalReferencePosition(lref: ReferencePosition): ReferencePosition | undefined {
+        const segment = lref.getSegment();
+        if (segment) {
+            const removedRefs = segment?.localRefs?.removeLocalRef(lref);
+            if (removedRefs !== undefined) {
+                this.blockUpdatePathLengths(segment.parent, TreeMaintenanceSequenceNumber,
+                    LocalClientId);
+            }
+            return removedRefs;
+        }
+    }
+    public createLocalReferencePosition(
+        segment: ISegment, offset: number, refType: ReferenceType, properties: PropertySet | undefined,
+        client: Client,
```

Comment:
This is adding the first reference to Client from MergeTree. I had assumed that was intentional - that MergeTree should not be aware of Client. Would it be cleaner to only have the creation API on client and make this an internal method called only by client that takes a LocalReference?

Reply:
this is just staging for back compat. i the next release we will no longer remove client which is deprecated.